### DISCUSSION TITLE
pocketbase relationship field mandatory need values during updates

### DISCUSSION THREAD

**Discussion by Jeyanthinath** [pocketbase/pocketbase#3223] [Q&A]
I have a use case for a blog.

The table fields are these.
id, name, domain, tags, *author (relation field with user), status

1. I want to update only the name field, but why I need to send the entire data of mandatory fields, can't I just send name data and it update that alone? 
2. as the author is a relationship field, I kept it mandatory for creation, but I don't want to send the data for this field every time for update.

Am I doing things completely wrong, is there any way I don't use author relationship field in this table but establish a relationship on who created the row (currently I am using it to make sure, the owner of the blog updates the meta data)

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand your questions, please elaborate or provide a code sample with the desired outcome (pseudo-code is also ok).

In general, you can send an update request with only the fields you want to update:

```js
await pb.collection("blog").update("RECORD_ID", {
    name: "new name"
})
```

**Reply by Jeyanthinath** [reply]
To put it in simple words, I have relationship called owner (which connects to user table) for every blog record which is being inserted (to identify the owner I am using that field , and I marked that field as manadatory)

| Name | * Owner (mandatory) |
|--------|--------|
| Blog 1 Title |  test@test.com |
| Blog 2 title | test1@test.com | 


without using relationship field is there a possibility to figure out **who created that record**? 

The problem I am facing is, as it is a mandatory field, I need to send the **owner** to the update function as mandatory, which seems anti-pattern, as the owner will be set only during creation of the record.

**Reply by ganigeorgiev** [reply]
> ... as it is a mandatory field, I need to send the owner to the update function as mandatory

This is incorrect.
If you have a relation field "owner" marked as nonemtpy you'll need to provide the owner id only during create.
When updating the record you don't have to specify again the owner id if you are not going to change it (_see the above code example_).
